contracts/contracts/strategies/GeneralizedUniswapV3Strategy.sol

```diff
@@ -0,0 +1,850 @@
+    }
+
+    /**
+     * @notice Change the address of the operator
+     * @dev Can only be called by the Governor
+     * @param _operator The new value to be set
+     */
+    function setOperator(address _operator) external onlyGovernor {
+        _setOperator(_operator);
+    }
+
+    function _setOperator(address _operator) internal {
+        require(_operator != address(0), "Invalid operator address");
+        operatorAddr = _operator;
+        emit OperatorChanged(_operator);
+    }
+
+    /**
+     * @notice Change the reserve strategies of the supported assets
+     * @param _token0ReserveStrategy The new reserve strategy for token0
+     * @param _token1ReserveStrategy The new reserve strategy for token1
+     */
+    function setReserveStrategy(
+        address _token0ReserveStrategy,
+        address _token1ReserveStrategy
+    ) external onlyGovernorOrStrategistOrOperator nonReentrant {
+        _setReserveStrategy(_token0ReserveStrategy, _token1ReserveStrategy);
+    }
+
+    /**
+     * @notice Change the reserve strategies of the supported assets
+     * @dev Will throw if the strategies don't support the assets
+     * @param _token0ReserveStrategy The new reserve strategy for token0
+     * @param _token1ReserveStrategy The new reserve strategy for token1
+     */
+    function _setReserveStrategy(
+        address _token0ReserveStrategy,
+        address _token1ReserveStrategy
+    ) internal {
+        require(
+            IStrategy(_token0ReserveStrategy).supportsAsset(token0),
+            "Invalid Reserve Strategy"
+        );
+        require(
+            IStrategy(_token1ReserveStrategy).supportsAsset(token1),
+            "Invalid Reserve Strategy"
+        );
+
+        reserveStrategy[token0] = _token0ReserveStrategy;
+        reserveStrategy[token1] = _token1ReserveStrategy;
+
+        emit ReserveStrategiesChanged(
+            _token0ReserveStrategy,
+            _token1ReserveStrategy
+        );
+    }
+
+    /***************************************
+            Deposit/Withdraw
+    ****************************************/
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == token0 || _asset == token1, "Unsupported asset");
+        IVault(vaultAddress).depositForUniswapV3(_asset, _amount);
+        // Not emitting Deposit event since the Reserve strategy would do so
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function depositAll() external override onlyVault nonReentrant {
+        _depositAll();
+    }
+
+    /**
+     * @notice Deposits all undeployed balances of the contract to the reserve strategies
+     */
+    function _depositAll() internal {
+        uint256 token0Bal = IERC20(token0).balanceOf(address(this));
+        uint256 token1Bal = IERC20(token1).balanceOf(address(this));
+        if (token0Bal > 0) {
```

Comment:
This could actually be greater than zero. If depositing dust costs $20 in gas, we do it once a day, and we earn 3% on the dust, then my math (which may be wrong) says we would still be better off keeping up to $240,000 dust in a Strat. This may be excessive, but something like $40K dust should be fine.

Reply:
Agreed, I guess we could also probably make it a configurable variable, if we wanna tweak it in future